config/routes.rb

```diff
@@ -1,11 +1,11 @@
 Rails.application.routes.draw do
-  get 'expenses/create'
-
   devise_for :users, controllers: { omniauth_callbacks: :omniauth_callbacks }
   root controller: :welcome, action: :index
-  get 'dashboard', to: 'welcome#dashboard'
-  resources :groups, only: [:new, :create, :show] do
-    resources :expenses, only: [:create]
-  end
 
+  authenticate :user do
```

Comment:
Interesting. Where did you find this solution?

Reply:
https://github.com/plataformatec/devise/wiki/How-To:-Define-resource-actions-that-require-authentication-using-routes.rb

Reply:
we ran in to some trouble with our controllers. Bit of a setup miss. Then we decided to find another way. And we found this one

Reply:
As long as you understand that every route is restricted using this method

Reply:
Yup! Because the only page we want people to be able to access is the welcome page, and everything else is restricted to logged in members